### PR TITLE
Non standard native attribute change.

### PR DESCRIPTION
**Pull request by firatsarlar**
Using native attribute in a tag is non standard.
This PR is about to use dataset-native instead of native attribute.
I may also note another PR : https://github.com/preactjs/preact/pull/2578

### CODE REVIEW

**Comment by firatsarlar** [comment]
Keeping native attribute check would be wiser for backward compatibility.

**Comment by Kangaroux** [comment]
@firatsarlar Any chance we can get this merged?

**Comment by firatsarlar** [comment]
@Kangaroux I think it will wait for a review process for 2-3 years .
Merging is not up to me.

File: src/index.js [hunk 1]
```diff
@@ -113,7 +113,7 @@ function delegateLinkHandler(e) {
 	let t = e.target;
 	do {
 		if (String(t.nodeName).toUpperCase()==='A' && t.getAttribute('href')) {
-			if (t.hasAttribute('native')) return;
+			if (t.hasAttribute('dataset-native') || t.hasAttribute('native')) return;
```
Comment (developit) [inline src/index.js:116]:
```suggestion
			if (t.hasAttribute('data-native') || t.hasAttribute('native')) return;
```

**Review by developit — DISMISSED** [review]
The attribute would be `data-native`.

I'd be interested to understand why this is seen as important. Nonstandard attributes are used extensively on the web, and there are no known performance or compatibility reasons to avoid them. They can cause issues with static HTML validators, however this is irrelevant for the `<a native>` attribute since its behavior is exclusively client-side.

**Comment by developit** [comment]
I would prefer to find a different mechanism for denoting links that should bypass client-side routing, if the attribute is cumbersome. There are multiple options, but this is my preferred one and is already support:

```js
<a target="_self" href="/">I will bypass client-side routing</a>
// ^^^^^^^^^^^^^^  DOM standard target attribute, and already respected by preact-router.
```

**Comment by Kangaroux** [comment]
Really what this PR should be for is adding support for Typescript, since `native` doesn't appear anywhere in the type definitions. Using `data-native` works since TS won't throw an error but it isn't the ideal solution.

**Comment by developit** [comment]
I believe that could be added [here](https://github.com/preactjs/preact-router/blob/5fb140e65241bdc42e06b7b7f0d66ff85af9a34e/index.d.ts#L68)

File: README.md [hunk 2]
```diff
@@ -144,10 +144,10 @@ render(
 
 Sometimes it's necessary to bypass preact-router's link handling and let the browser perform routing on its own.
 
-This can be accomplished by adding a `native` boolean attribute to any link:
+This can be accomplished by adding a `dataset-native` boolean attribute to any link:
 
 ```html
-<a href="/foo" native>Foo</a>
+<a href="/foo" dataset-native>Foo</a>
```
Comment (marvinhagemeister) [inline README.md:150]:
```suggestion
<a href="/foo" data-native>Foo</a>
```

**Review by marvinhagemeister — APPROVED** [review]

**Pull request merged**